templates/aws/awsdeploy.yml

```diff
@@ -0,0 +1,287 @@
+                - apprunner.amazonaws.com
+                - tasks.apprunner.amazonaws.com
+                - build.apprunner.amazonaws.com
+                - cloudformation.amazonaws.com
+            Action: 'sts:AssumeRole'
+      Policies:
+        - PolicyName: SMPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'secretsmanager:GetSecretValue'
+                Resource: !Join
+                  - ''
+                  - - !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:'
+                    - !Ref UniqueSolutionPrefix
+                    - '-*'
+        - PolicyName: AmazonElasticContainerRegistryPublicReadOnly
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'ecr-public:GetAuthorizationToken'
+                  - 'sts:GetServiceBearerToken'
+                  - 'ecr-public:BatchCheckLayerAvailability'
+                  - 'ecr-public:GetRepositoryPolicy'
+                  - 'ecr-public:DescribeRepositories'
+                  - 'ecr-public:DescribeRegistries'
+                  - 'ecr-public:DescribeImages'
+                  - 'ecr-public:DescribeImageTags'
+                  - 'ecr-public:GetRepositoryCatalogData'
+                  - 'ecr-public:GetRegistryCatalogData'
+                Resource: '*'
+        - PolicyName: AWSAppRunnerFullAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action: 'iam:CreateServiceLinkedRole'
+                Resource: 'arn:aws:iam::*:role/aws-service-role/apprunner.amazonaws.com/AWSServiceRoleForAppRunner'
+                Condition:
+                  StringLike:
+                    iam:AWSServiceName: 'apprunner.amazonaws.com'
+              - Effect: Allow
+                Action: 'iam:PassRole'
+                Resource: '*'
+                Condition:
+                  StringLike:
+                    iam:PassedToService: 'apprunner.amazonaws.com'
+              - Sid: AppRunnerAdminAccess
+                Effect: Allow
+                Action: 'apprunner:*'
+                Resource: '*'
+        - PolicyName: AWSAppRunnerServicePolicyForECRAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'ecr:GetDownloadUrlForLayer'
+                  - 'ecr:BatchGetImage'
+                  - 'ecr:DescribeImages'
+                  - 'ecr:GetAuthorizationToken'
+                  - 'ecr:BatchCheckLayerAvailability'
+                Resource: '*'
+        - PolicyName: ECR-all
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'ecr:*'
```

Comment:
Those rules are not compliant with least privileges principle. 
The runner should be only allowed to read the registries....

Since ``AWSAppRunnerServicePolicyForECRAccess`` and ``AmazonElasticContainerRegistryPublicReadOnly`` are already present, I suggest to remove this policy.